qa/tasks/cephfs/test_nfs.py

```diff
@@ -334,6 +365,53 @@ def _write_to_read_only_export(self, pseudo_path, port, ip):
         else:
             self.fail('expected write to a read-only export to fail')
 
+    def _create_cluster_with_fs(self, fs_name, mnt_pt=None):
+        """
+        create a cluster along with fs and mount it to the path supplied
+        :param fs_name: name of CephFS volume to be created
+        :param mnt_pt: mount fs to the path
+        """
+        self._test_create_cluster()
+        self._cmd('fs', 'volume', 'create', fs_name)
+        with contextutil.safe_while(sleep=5, tries=30) as proceed:
+            while proceed():
+                output = self._cmd(
+                    'orch', 'ls', '-f', 'json',
+                    '--service-name', f'mds.{fs_name}'
+                )
+                j = json.loads(output)
+                if j[0]['status']['running']:
+                    break
+        if mnt_pt:
+            tries = 3
+            while True:
+                try:
+                    self.ctx.cluster.run(args=['sudo', 'ceph-fuse', mnt_pt])
+                    break
+                except CommandFailedError:
+                    if tries:
+                        tries -= 1
+                        time.sleep(3)
+                        continue
+                    raise
```

Comment:
Dropping this `raise` will throw the exception and also will make the use of `continue` redundant.

Reply:
if i remove this raise then the program execution will continue and try `self.ctx.cluster.run(args=['sudo', 'chmod', '1777', mnt_pt])` which is not expected, btw this is outside `contextutil.safe_while` block so this needs to be raised

Reply:
this whole block has been now replaced with `safe_while` (was using `while()` previously), so better handling and no more cluttered code, thus resolving this conversation.